pkg/github/structs.go

```diff
@@ -224,7 +234,7 @@ func (e PullRequestEvent) GetHeadBranch() string {
 	return e.GetPullRequest().GetHead().GetRef()
 }
 
-func (e PullRequestEvent) IsValid() bool {
+func (e PullRequestEvent) IsValid(c *config.Config) bool {
 	if !utils.StrInSlice(getAuthorizedPullRequestActions(), e.GetAction()) {
```

Comment:
why don't you check `e.GetCheckRun().GetApp().GetID()` for `PullRequestEvent` ?

Reply:
The field does not exist in PullRequestEvent (and the event happening on pull requests are unique and not by checksuite which was my duplicate problem)